go/private/rules/cgo.bzl

```diff
@@ -180,20 +192,29 @@ def cgo_configure(go, srcs, cdeps, cppopts, copts, cxxopts, clinkopts):
         clinkopts = clinkopts,
     )
 
-def _cc_libs(target):
+def _cc_libraries_to_link(target):
+    libraries_to_link = as_iterable(target[CcInfo].linking_context.libraries_to_link)
+    return libraries_to_link
+
+def _cc_libs_for_target(target):
     # Copied from get_libs_for_static_executable in migration instructions
     # from bazelbuild/bazel#7036.
-    libraries_to_link = as_iterable(target[CcInfo].linking_context.libraries_to_link)
+    libraries_to_link = _cc_libraries_to_link(target)
     libs = []
     for library_to_link in libraries_to_link:
-        if library_to_link.static_library != None:
-            libs.append(library_to_link.static_library)
-        elif library_to_link.pic_static_library != None:
-            libs.append(library_to_link.pic_static_library)
-        elif library_to_link.interface_library != None:
-            libs.append(library_to_link.interface_library)
-        elif library_to_link.dynamic_library != None:
-            libs.append(library_to_link.dynamic_library)
+        libs.extend(_cc_libs(library_to_link))
+    return libs
+
+def _cc_libs(library_to_link):
+    libs = []
```

Comment:
> Why return a list from this function instead of a single value?

When I extracted that from the original source, I misread the code -- I thought I had read a set of four distinct `if` conditions (implying that `library_to_link` might have multiple files that need to be handled), _not_ one chain of `if ... elif`. I'll change that.